templates/WAM_APP_FRED/test_map_layout.html

```diff
@@ -472,8 +489,234 @@ <h2 class="tabs-title__head">Feedin-Time Series</h2>
+                        onEachFeature: enablePowPlPopup,
+                        pointToLayer: createCircles
+                      });
+                      //map.addLayer(layer)
+                      if (regionPowerPlantgroup.getLayers().length === 0) {
+                        regionPowerPlantgroup.addLayer(layer)
+                      } 
+                      else {
+                        regionPowerPlantgroup.clearLayers();
+                      }
+
+                    }
+                  });
+                }
+              
+              // add all power plants from url to map
+              function addPowerPlantToMap(){
+
+              }
+              
+              $.ajax({
+                    url: ppr_location,
+                    dataType: 'json',
+                    async: true,
+                    success: function(powerplant){
+
+                        var layer = new L.geoJSON(powerplant, {
+                            onEachFeature: enablePowPlPopup,
+                            pointToLayer: createCircles
+                        });
+                        map.addLayer(layer)
+                        }
+                });
+
+
+
+                function onPowerPlantDataPointClick (event) {
+                  var marker = event.sourceTarget;
+                  // console.log(marker);
+                  var leaflet_id = L.stamp(marker);
+                  var pp_id = marker.feature.id;
+                  console.log(pp_id)
+                  $.ajax({
+                      type: 'POST',
+                      url: ppr_contend,
+                      headers: {'X-CSRFToken': '{{ csrf_token }}'},
+                      data: {
+                      'leaflet_id': leaflet_id,
+                      'pp_id': pp_id
+                      },
+                      success: function(thispopupcontend){
+                        displayPowerPlantDataPoint(thispopupcontend, marker)
+                      }
+                  });
+                }
+
+                function displayPowerPlantDataPoint(feature, marker) {
+                  if (feature.property) {
+                    var keys = Object.keys(feature.property);
+                    var content = '<table class="selection_detail">';
+                    
+                    for (let i = 0, len = keys.length; i < len; i++) {
+                      content = content + '<tr><td align="right"><b>' + keys[i] + '</b>:</td><td>' + feature.property[keys[i]] + '</td></tr>'
+                    }
+                    content = content + '</table>';
+                    
+                    console.log(marker);
+                    marker._popup.setContent(content);
+
+                    // is this working????
+                    addPowerPlantMarker(feature, marker);
+
+                  }
+                }
+
+              // add icon on to clicked circelmarker 
+              function addPowerPlantMarker (feature, marker){
```

Comment:
why not store the icon information in a dict ?

Reply:
the geojson feature is like a dict in opinion